packages/react-client/src/DeviceManager.ts

```diff
@@ -47,16 +47,16 @@ export class DeviceManager
 
   constructor(deviceType: "audio" | "video", defaultConfig?: DeviceManagerConfig) {
     super();
-    this.storageConfig = this.createStorageConfig(defaultConfig?.storage);
+    this.storageConfig = this.createStorageConfig(deviceType, defaultConfig?.storage);
 
     this.deviceType = deviceType;
 
     this.constraints = toMediaTrackConstraints(defaultConfig?.trackConstraints ?? true);
   }
 
-  private createStorageConfig(storage?: boolean | StorageConfig): StorageConfig | null {
+  private createStorageConfig(deviceType: "audio" | "video", storage?: boolean | StorageConfig): StorageConfig | null {
     if (storage === false) return null;
-    if (storage === true || storage === undefined) return getLocalStorageConfig(this.deviceType);
+    if (storage === true || storage === undefined) return getLocalStorageConfig(deviceType);
```

Comment:
**question:** why is this change needed? if the device type was undefined, then maybe we should set it before invoking `this.createStorageConfig` in line 50.

Reply:
Yeah, you're right. I thought about it, and I think this approach is safer because it's not dependent on the order of function invocations in the constructor. It makes createStorageConfig a pure function, so I decided to write it this way. TS disappointed me because it allows private readonly deviceType: "audio" | "video" to be undefined.

Reply:
spox